### DISCUSSION TITLE
PHP 8.1 interbase problem

### DISCUSSION THREAD

**Discussion by ssu1231** [FirebirdSQL/php-firebird#33] [Q&A]
Hi, 
I have a question about the drivers php-interbase for php 8.1 and Firebird, Windows x64? 
Are they available for download somewhere?

I checked version 8.0 and unfortunately it doesn't work :(

thank you in advance

Regards
S

**Comment by JPirene** [comment]
How to install? The Read.Me Doc not has information to install. Help me please!!!

**Reply by ssu1231** [reply]
Hi JPirene, here is a link with the relevant versions https://github.com/FirebirdSQL/php-firebird/releases I don't know what you want to do, but I use Xampp. I copy the appropriate php-interbase files there.

**Reply by JPirene** [reply]
Sorry, I didn't explain. 
I want connect to firebase using PHP 8.1.*  in the xampp

I'm trying to use php_pdo_firebase.dll (extension) but when using the function "PDO::pdo_drivers()" the list does not show firebird. And then I found this repository, it seems to be a solution.

Which directory should I copy?
Which Files?
Should I change php.ini? Where?

**Reply by MartinKoeditz** [reply]
Hi,

this is the repository of the Firebird PHP driver. At this point we don't support the Firebird PHP PDO driver.

**Reply by mrotteveel** [reply]
> I'm trying to use php_pdo_firebase.dll (extension) but when using the function "PDO::pdo_drivers()" the list does not show firebird. And then I found this repository, it seems to be a solution.

php_pdo_firebase.dll, if it existed, would not be for Firebird, but for Google Firebase. You need php_pdo_firebird.dll, and that is a different driver than this one.

XAMPP comes with php_pdo_firebird.dll, but has it disabled by default. To be able to use PDO Firebird in XAMPP, you need to uncomment `extension=pdo_firebird` (by removing the `;` before it) in php.ini and have a suitable fbclient.dll installed, e.g. using the Firebird installer).

**Reply by JPirene** [reply]
Wow, I commented wrong!!! I have php_pdo_firebird.dll and no php_pdo_firebase.dll, I resolved my trouble reinstalling firebird 2.5 on version 64bit and select box on installation to copy dll on my system folder. Aplication Xampp work at arquiteture 64bit and Firebird server must has install in same arquiteture.

Thanks for the help!

**Reply by mrotteveel** [reply]
> Aplication Xampp work at arquiteture 64bit and Firebird server must has install in same arquiteture.

That is not correct, the Firebird server doesn't need to be the same architecture as PHP/XAMPP, but you need to install the Firebird client library (fbclient.dll) of the right architecture. Selecting the client installation option in the 64-bit installer would have worked as well.

**Comment by juliogyn** [comment]
Hi, maybe someone can help me. I have xampp and php 8 installed, and I can't get php to see the firebird client. I have a fresh firebird 3-32bit installed and I keep getting the message "PHP Warning:  PHP Startup: Unable to load dynamic library 'pdo_firebird' (tried: C:\xampp\php\ext\pdo_firebird (The specified module could not be found), C:\xampp\php\ext\php_pdo_firebird.dll (%1 is not a valid Win32 application)) in Unknown on line 0"

**Reply by ssu1231** [reply]
Hi, maybe you should try to change it in php.ini:
 
;extension=pdo_firebird
extension=php_interbase.dll

**Reply by ssu1231** [reply]
or https://stackoverflow.com/questions/48923733/php-startup-unable-to-load-dynamic-library-php-pdo-firebird

**Reply by juliogyn** [reply]
After trying many things I found one solution, at least for now. I made a copy of the pdo_firebird.dll and fbclient.dll (64 bits) to windows\system32 and php loaded the pdo in phpinfo(). php now working and showing data
